cache/class-vary-cache.php

```diff
@@ -511,11 +521,16 @@ private static function set_cookies() {
 	 * @access  private
 	 */
 	private static function set_group_cookie() {
+		$group_string = self::stringify_groups();
+		if ( empty( $group_string ) ) {
+			return;
```

Comment:
Something to track in a separate issue: we may want to consider a `remove_user_from_group` and unsetting cookies when a user goes from being in a group with a cookie to being in no groups.